src/data/starredSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from "@reduxjs/toolkit"
+
+const starredSlice = createSlice({
+    name: 'starred',
+    initialState: {
+        starredMovies: []
+    },
+    reducers: {
+        starMovie: (state, action) => {
+            state.starredMovies = [action.payload, ...state.starredMovies]
```

Comment:
```suggestion
            state.starredMovies = [...state.starredMovies, action.payload]
```

The order in JavaScript destructuring is important